src/environment/environment.js

```diff
@@ -0,0 +1,127 @@
+/** @flow */
+import os from 'os';
+import { v4 } from 'uuid';
+import fs from 'fs-extra';
+import path from 'path';
+import npmInstall from '../utils/npm';
+import { loadScope } from '../scope';
+import { flattenDependencies } from '../scope/flatten-dependencies';
+import { BitId } from '../bit-id';
+import { Component } from '../consumer/component/consumer-component'
+import { BITS_DIRNAME } from '../constants';
+import { mergeAll } from 'ramda';
+
+const root = path.join(os.tmpdir(), 'bit');
+const currentPath = process.cwd();
+
+export default class Environment {
+  path: string;
+  component: Component;
+  componentsDependencies: ConsumerComponent[] = [];
+
+  constructor() {
+    this.path = path.join(root, v4());
+    console.log(this.path);
+  }
+
+  init(): Promise<*> {
+    return new Promise((resolve, reject) => {
+      fs.ensureDir(this.path, (err) => {
+        if (err) return reject(err);
+        return resolve();
+      });
+    });
+  }
+
+  importById(rawId: string): Promise<Object> {
+    // loadScope works for local scope and remote scope
+    return loadScope(currentPath).then((scope) => {
+      const bitId = BitId.parse(rawId, scope.name);
+      return scope.get(bitId)
+        .then((component) => {
+          component.component.write(this.path, true, true);
+          this.component = component.component;
+          return component;
+        });
+    });
+  }
+
+  writeToEnvironmentDir(componentDependencies: versionDependencies[]): Promise<Component[]> {
+    const componentsDir = path.join(this.path, BITS_DIRNAME);
+    const components = flattenDependencies(componentDependencies);
+
+    const bitDirForEnvironmentImport = (component: Component) => {
+      return path.join(
+        componentsDir,
+        component.box,
+        component.name,
+        component.scope,
+        component.version.toString(),
+      );
+    };
+
+    return Promise.all(components.map((component) => {
+      const bitPath = bitDirForEnvironmentImport(component);
+      return component.write(bitPath, true);
+    }));
+  }
+
+  importDependencies(): Promise<ConsumerComponent[]> {
+    if (!this.component) return Promise.reject();
```

Comment:
As discussed, this is how it was implemented in the first place, but I encountered some issues. 
I rewrote the implementation of using the Consumer along with the old implementation, so we could compare them.
The new Consumer implementation can be used as following:
```js
const environment = new Environment();
environment
    .init()
    .then(() => environment.initConsumer())
    .then(() => environment.importByConsumer(['bit.utils/array/diff']))
    .then(components => environment.npmInstallByConsumer(components));
```
Ignore the method names, they will be replaced. 
The first issue I have found is that it doesn't install the compiler nor the tester. I have also tried to run it from the command line: `bit-prod import bit.utils/array/diff -e -s`, but I got the same results. Am I missing something?

Another issue is the performance. By using a consumer it performs unnecessary steps such as writing to .bit/objects.

We can ignore the second issue for now, but the first one is a blocker.